src/Symfony/Component/Config/ConfigCache.php

```diff
@@ -11,22 +11,27 @@
 
 namespace Symfony\Component\Config;
 
-use Symfony\Component\Config\Resource\ResourceInterface;
-use Symfony\Component\Filesystem\Exception\IOException;
-use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Config\Resource\BCResourceInterfaceChecker;
+use Symfony\Component\Config\Resource\DefaultResourceChecker;
 
 /**
- * ConfigCache manages PHP cache files.
+ * ConfigCache caches arbitrary content in files on disk.
  *
- * When debug is enabled, it knows when to flush the cache
- * thanks to an array of ResourceInterface instances.
+ * When in debug mode, those metadata resources that implement
+ * \Symfony\Component\Config\Resource\SelfCheckingResourceInterface will
+ * be used to check cache freshness.
+ *
+ * During a transition period, also instances of
+ * \Symfony\Component\Config\Resource\ResourceInterface will be checked
+ * by means of the isFresh() method. This behaviour is deprecated since 2.8
+ * and will be removed in 3.0.
  *
  * @author Fabien Potencier <fabien@symfony.com>
+ * @author Matthias Pigulla <mp@webfactory.de>
  */
-class ConfigCache implements ConfigCacheInterface
+class ConfigCache extends ResourceCheckerConfigCache
```

Comment:
why putting the debug mode in a child class ? this makes it much harder to have a debug and non-debug mode for the cache, as it would mean changing the registration of checkers in the config cache, which is much harder than switching a boolean flag (and not supported by your compiler pass btw)